packages/webpack-cli/src/webpack-internal-types.ts

```diff
@@ -0,0 +1,176 @@
+   */
+  compression?: false | "gzip" | "brotli";
+
+  /**
+   * Algorithm used for generation the hash (see node.js crypto package).
+   */
+  hashAlgorithm?: string;
+
+  /**
+   * Time in ms after which idle period the cache storing should happen.
+   */
+  idleTimeout?: number;
+
+  /**
+   * Time in ms after which idle period the cache storing should happen when larger changes has been detected (cumulative build time > 2 x avg cache store time).
+   */
+  idleTimeoutAfterLargeChanges?: number;
+
+  /**
+   * Time in ms after which idle period the initial cache storing should happen.
+   */
+  idleTimeoutForInitialStore?: number;
+
+  /**
+   * List of paths that are managed by a package manager and contain a version or hash in its path so all files are immutable.
+   */
+  immutablePaths?: (string | RegExp)[];
+
+  /**
+   * List of paths that are managed by a package manager and can be trusted to not be modified otherwise.
+   */
+  managedPaths?: (string | RegExp)[];
+
+  /**
+   * Time for which unused cache entries stay in the filesystem cache at minimum (in milliseconds).
+   */
+  maxAge?: number;
+
+  /**
+   * Number of generations unused cache entries stay in memory cache at minimum (0 = no memory cache used, 1 = may be removed after unused for a single compilation, ..., Infinity: kept forever). Cache entries will be deserialized from disk when removed from memory cache.
+   */
+  maxMemoryGenerations?: number;
+
+  /**
+   * Additionally cache computation of modules that are unchanged and reference only unchanged modules in memory.
+   */
+  memoryCacheUnaffected?: boolean;
+
+  /**
+   * Name for the cache. Different names will lead to different coexisting caches.
+   */
+  name?: string;
+
+  /**
+   * Track and log detailed timing information for individual cache items.
+   */
+  profile?: boolean;
+
+  /**
+   * When to store data to the filesystem. (pack: Store data when compiler is idle in a single file).
+   */
+  store?: "pack";
+
+  /**
+   * Filesystem caching.
+   */
+  type: "filesystem";
+
+  /**
+   * Version of the cache data. Different versions won't allow to reuse the cache and override existing content. Update the version when config changed in a way which doesn't allow to reuse cache. This will invalidate the cache.
+   */
+  version?: string;
+}
+
+export interface AssetEmittedInfo {
+  content: Buffer;
+  source: Source;
+  compilation: Compilation;
+  outputPath: string;
+  targetPath: string;
+}
+type EntryDescriptionNormalized = any;
+
+export interface Argument {
+  description: string;
+  simpleType: "string" | "number" | "boolean";
+  multiple: boolean;
+  configs: ArgumentConfig[];
+}
+interface ArgumentConfig {
+  description: string;
+  negatedDescription?: string;
+  path: string;
+  multiple: boolean;
+  type: "string" | "number" | "boolean" | "path" | "enum" | "RegExp" | "reset";
+  values?: any[];
+}
+
+export type EntryOptions = { name?: string } & Omit<EntryDescriptionNormalized, "import">;
```

Comment:
We should fix it in webpack and export them, no need to duplicate, it is just mistake on webpack side

Reply:
Thank you, I've submitted a [PR](https://github.com/webpack/webpack/pull/15377)

Reply:
I believe we need to update webpack here.